netbox/netbox/settings.py

```diff
@@ -811,6 +812,15 @@ def _setting(name, default=None):
             f"__init__.py file and point to the PluginConfig subclass."
         )
 
+    # Validate version compatibility and user-provided configuration settings and assign defaults
+    if plugin_name not in PLUGINS_CONFIG:
+        PLUGINS_CONFIG[plugin_name] = {}
+    try:
+        plugin_config.validate(PLUGINS_CONFIG[plugin_name], RELEASE.version)
```

Comment:
We can simplify this a bit:
```suggestion
    try:
        plugin_config.validate(PLUGINS_CONFIG.get(plugin_name, {}), RELEASE.version)
```

Reply:
Actually I don't think this can be done this way. Trying to `validate` in one fell swoop like this calls `get_plugin_config` which raises an `ImproperlyConfigured` error which I'm not trapping in this try-except (intentionally), and that fires before it checks whether the key is in the dict. It's weird and ugly but the way I had it is the only way to do it without changing `get_plugin_config` too with even more abstractions.